### ISSUE TITLE
Model is validating, However AP/AR scores don't budge from 0

### ISSUE THREAD

**Issue by neel04** [open-mmlab/mmdetection#6359]
This is a migration of #6253; as it wasn't appropriate for the level of complexity of this issue

**Describe the bug**
Despite the validation set being loaded in, and the model overfitting (but not severely) we still have AP/AR scores for validation to be `0` and `-1`s

**Reproduction**

Specified in issue chain

2. Did you make any modifications on the code or config? Did you understand what you have modified? `No`
3. What dataset did you use? `Custom COCO dataset`

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.

```py
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
sys.platform: linux
Python: 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 16:07:37) [GCC 9.3.0]
CUDA available: True
GPU 0: Tesla P100-PCIE-16GB
CUDA_HOME: /usr/local/cuda
NVCC: Build cuda_11.0_bu.TC445_37.28845127_0
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.7.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.2-Product Build 20210312 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.1
OpenCV: 4.5.2
MMCV: 1.3.15
MMCV Compiler: GCC 7.5
MMCV CUDA Compiler: 11.0
MMDetection: 2.17.0+
```

2. `Kaggle Env`

**Error traceback**
Specified in the previously linked chain #6253 that I am unable to get the model to validate properly.

**Comment by neel04** [comment]
I trried a few fixes, but the evaluation metric remain same. however, during inference, using the standard way of inferencing in the docs:-
```py
from mmdet.apis import init_detector, inference_detector, show_result_pyplot

config = "/kaggle/working/Swin-Transformer-Object-Detection/configs/cascade_rcnn/cascade_mask_rcnn_swin_small_patch4_window7_mstrain_480-800_giou_4conv1f_adamw_3x_coco.py"

checkpoint = '/kaggle/working/results/epoch_1.pth'
model = init_detector(config, checkpoint, device='cuda:0')

img = '../input/new-passion-fruit-dataset/augmented_train_coco/data/ID_8YY1OE1J.jpg' #test image here
result = inference_detector(model, img)
print(result)
show_result_pyplot(model, img, result, score_thr=0.3)
```
I get 
```py
result = [array([[1.90793549e+02, 1.66902267e+02, 3.34713013e+02, 4.50667297e+02,
        1.41145363e-01],
       [2.13450180e+02, 1.81740280e+02, 4.50778137e+02, 4.26466339e+02,
        1.09936081e-01],
       [1.59150894e+02, 1.10531586e+02, 4.25755890e+02, 4.99718140e+02,
        9.62275565e-02],
       [2.17322464e+02, 2.47503555e+02, 3.59565338e+02, 4.66293488e+02,
        9.35899317e-02],
       [1.58171066e+02, 2.08359604e+02, 3.93465088e+02, 3.74480103e+02,
        7.82584697e-02],
       [1.95506683e+02, 8.72390594e+01, 3.77070343e+02, 3.96860901e+02,
        7.32497200e-02],
       [1.63192719e+02, 2.66339264e+02, 4.10874268e+02, 4.34276825e+02,
        5.52976802e-02]], dtype=float32), array([[1.7443633e+02, 2.1092377e+02, 3.5021484e+02, 4.1769016e+02,
        1.2604180e-01],
       [1.5582779e+02, 1.2804681e+02, 4.6357590e+02, 4.2469528e+02,
        1.2184056e-01],
       [1.4296977e+02, 1.7244580e+02, 3.6683313e+02, 5.1200000e+02,
        9.9554576e-02],
       [2.2514923e+02, 2.0445786e+02, 4.1564383e+02, 4.1223254e+02,
        9.5624447e-02],
       [1.5381857e+02, 1.8421429e+02, 3.8451376e+02, 3.4292407e+02,
        6.9557719e-02],
       [4.5520470e+01, 1.4345824e+02, 5.1200000e+02, 5.0660718e+02,
        6.8989336e-02],
       [7.1513260e+01, 2.1142796e+02, 4.0034930e+02, 3.9143414e+02,
        6.5558597e-02],
       [2.4822406e+02, 2.4561777e+02, 3.8182993e+02, 4.7127112e+02,
        5.7746951e-02],
       [9.1152451e+01, 1.1780274e+00, 3.9961667e+02, 4.7162915e+02,
        5.5406727e-02]], dtype=float32), array([[1.8953935e+02, 1.6757422e+02, 3.3653064e+02, 4.4860159e+02,
        1.5313146e-01],
       [1.5570259e+02, 1.3058449e+02, 4.6179132e+02, 4.2391290e+02,
        1.3083327e-01],
       [2.5434015e+02, 1.8519438e+02, 3.9406570e+02, 4.7249445e+02,
        1.2200830e-01],
       [1.5400204e+02, 2.0901758e+02, 3.9439624e+02, 3.7442908e+02,
        7.4130602e-02],
       [9.8904640e+01, 9.6329140e+01, 3.8437308e+02, 5.0941507e+02,
        6.0153686e-02],
       [1.6016638e+02, 2.6576160e+02, 4.1135437e+02, 4.3359552e+02,
        5.0923157e-02]], dtype=float32)
```
but the image output from the inference is empty 😢 anyone have any ideas what the list above means?

**Comment by neel04** [comment]
Interestingly, changing the `score_threshold` allows us to see better predictions - does that imply my model is very uncertain as to what the boxes and score are? 🤔

**Comment by jshilong** [comment]
> Interestingly, changing the `score_threshold` allows us to see better predictions - does that imply my model is very uncertain as to what the boxes and classes are? thinking By manually playing, I am able to converge on correct predictions but they are low confidence... so is this indicative of overfitting?

It may depend on your dataset, I suggest checking the code to parse your annotations is right and it is best to visualize the bounding box on your input to make sure you put correct data to the model.
